src/components/Interpretations/common/UserMention/UserMentionWrapper.js

```diff
@@ -0,0 +1,213 @@
+                setCapturedText(filterValue)
+            } else if (filterValue.length === 0) {
+                setCapturedText('')
+                clear()
+            }
+        }
+    }
+
+    // event bubbles up from the wrapped input/textarea
+    const onKeyDown = ({ key, target }) => {
+        setElement(target)
+
+        const { selectionStart } = target
+
+        if (!captureText && key === '@') {
+            setCaptureText(true)
+            setCaptureStartPosition(selectionStart + 1)
+        } else if (captureText) {
+            if (
+                key === ' ' ||
+                (key === 'Backspace' && selectionStart <= captureStartPosition)
+            ) {
+                reset()
+            } else if (users.length) {
+                switch (key) {
+                    case 'Enter':
+                        event.preventDefault()
+                        if (selectedUserIndex >= 0) {
+                            onSelect(users[selectedUserIndex])
+                        }
+                        break
+                    case 'ArrowDown':
+                        event.preventDefault()
+
+                        if (selectedUserIndex < users.length - 1) {
+                            setSelectedUserIndex(selectedUserIndex + 1)
+                        }
+
+                        break
+                    case 'ArrowUp':
+                        event.preventDefault()
+
+                        if (selectedUserIndex > 0) {
+                            setSelectedUserIndex(selectedUserIndex - 1)
+                        }
+
+                        break
+                    default:
+                    // other key strokes, typically the text typed
+                    // the onInput event handler set on the input element is triggering the user lookup
+                }
+            }
+        }
+    }
+
+    const onSelect = user => {
+        const originalValue = element.value
+        const newValue = `${originalValue.slice(
+            0,
+            captureStartPosition - 1
+        )}${originalValue
+            .slice(captureStartPosition - 1)
+            .replace(/^@\w*/, `@${user.userCredentials.username} `)}`
+
+        reset()
+
+        // typically for connected components we want the state to be updated too
+        // but the logic belongs to the wrapped component, so we just invoke the supplied callback
+        if (onUserSelect) {
+            onUserSelect(newValue)
+        }
+
+        // need to refocus on the input/textarea
+        element.focus()
+
+        // position the cursor at the end
+        setTimeout(() => element.setSelectionRange(-1, -1), 0)
+    }
+
+    const onClick = user => (_, event) => {
+        if (event) {
+            event.stopPropagation()
+            event.preventDefault()
+        }
+
+        onSelect(user)
+    }
+
+    return (
+        <div onKeyDown={onKeyDown} onInput={onInput}>
+            {children}
```

Comment:
Cool, this is the one point in the TODO section in the PR.
I tried an approach I read on Stackoverflow where it seemed possible to get the cursor coordinates from the cursor position using the selection info of the textarea, but it didn't work for me, although I didn't investigate much.

The one doubt is still about the popper being moved in some cases because there isn't enough space in the viewport, for example if the `@` is at the end of the textarea width, or close to the bottom of the textarea and with little or no space under the texarea.

Reply:
Reg. your doubt, I think if we ensure the virtualElement has correct dimensions, i.e. also a width and height, then the popper's positioning strategies should work correctly. If not, our Popper component even accepts a `modifiers` prop which we could use to implement a different strategy.